api/provider/aws/environment_update.go

```diff
@@ -52,3 +126,78 @@ func (e *EnvironmentProvider) updateASGSize(autoScalingGroupName string, minSize
 
 	return nil
 }
+
+func (e *EnvironmentProvider) createIngressInput(sourceGroupID, destGroupID string) error {
+	groupPair := &ec2.UserIdGroupPair{}
+	groupPair.SetGroupId(destGroupID)
+
+	permission := &ec2.IpPermission{}
+	permission.SetIpProtocol("-1")
+	permission.SetUserIdGroupPairs([]*ec2.UserIdGroupPair{groupPair})
+
+	ingressInput := &ec2.AuthorizeSecurityGroupIngressInput{}
+	ingressInput.SetGroupId(sourceGroupID)
+
+	ingressInput.SetIpPermissions([]*ec2.IpPermission{permission})
+
+	if _, err := e.AWS.EC2.AuthorizeSecurityGroupIngress(ingressInput); err != nil {
+		if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() == "InvalidPermission.Duplicate" {
+			log.Printf("[WARN] skipping environment link since rule for security group '%s' already exists\n", destGroupID)
+			return nil
+		}
+
+		return err
+	}
+
+	return nil
+}
+
+func (e *EnvironmentProvider) setLinkTags(environmentID string, links []string) error {
+	uniqueLinks := []string{}
```

Comment:
I thought the same thing when I found `StringSlice` in the `sort` package. But that seemed overkill too. But as you suggested it, it would make it better. Helper methods like Contains doesn't really belong anywhere (on a sidenote: kind of shocking that none go standard package doesn't contain a `contains` on an unordered list).

Reply:
✅

Reply:
i know right? oh well 🤷‍♂️